src/moonlink/src/storage/mooncake_table/tests.rs

```diff
@@ -1142,38 +1219,1290 @@ async fn test_streaming_begin_flush_abort_end_flush_multiple() {
+
+    // Step 7: Force data compaction
+    let created = table.create_snapshot(SnapshotOption {
+        uuid: uuid::Uuid::new_v4(),
+        force_create: true,
+        skip_iceberg_snapshot: true,
+        index_merge_option: MaintenanceOption::Skip,
+        data_compaction_option: MaintenanceOption::ForceRegular, // Force data compaction
+    });
+
+    assert!(created, "Mooncake snapshot should be created");
+
+    sync_mooncake_snapshot(&mut table, &mut event_completion_rx).await;
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_stream_commit_with_pending_flush_deletion_remapping_bug() -> Result<()> {
+    let context = TestContext::new("stream_commit_pending_flush_deletion_bug");
+    let mut table = test_table(
+        &context,
+        "stream_commit_pending_flush_deletion_bug",
+        IdentityProp::Keys(vec![0]),
+    )
+    .await;
+    let (event_completion_tx, mut event_completion_rx) = mpsc::channel(100);
+    table.register_table_notify(event_completion_tx).await;
+
+    // Step 1: Add some initial data and flush it to create a data file
+    append_rows(&mut table, vec![test_row(1, "Initial", 20)])?;
+    table.commit(1);
+    flush_table_and_sync(&mut table, &mut event_completion_rx, 2).await?;
+
+    // Step 2: Start a streaming transaction
+    let xact_id = 1;
+    table.append_in_stream_batch(test_row(2, "Stream", 21), xact_id)?;
+
+    // Step 3: Delete a row from the committed data within the streaming transaction
+    // This adds the deletion to committed_deletion_log
+    table
+        .delete_in_stream_batch(test_row(1, "Initial", 20), xact_id)
+        .await;
+
+    // Step 4: Flush the streaming transaction (creates disk slice)
+    // The disk slice now contains the deletion remapping info
+    let disk_slice =
+        flush_stream_and_sync_no_apply(&mut table, &mut event_completion_rx, xact_id, Some(4))
+            .await
+            .expect("Disk slice should be present");
+
+    // Step 5: Commit the streaming transaction
+    // This processes deletions and adds them to committed_deletion_log
+    // but the disk slice is still pending to be integrated
+    table.commit_transaction_stream_impl(xact_id, 3)?;
+
+    // Step 6: Apply the stream flush result
+    // This adds the disk slice to snapshot_task for later processing
+    table.apply_stream_flush_result(xact_id, disk_slice);
+
+    // Step 7: Create a snapshot to trigger integrate_disk_slices
+    // Without the fix, this would fail because deletions in committed_deletion_log
+    // get remapped but not applied to batch_deletion_vector
+    let created = table.create_snapshot(SnapshotOption {
+        uuid: uuid::Uuid::new_v4(),
+        force_create: true,
+        skip_iceberg_snapshot: true,
+        index_merge_option: MaintenanceOption::Skip,
+        data_compaction_option: MaintenanceOption::Skip,
+    });
+    assert!(created, "Mooncake snapshot should be created");
+
+    // This should succeed with the fix, but would fail without it
```

Comment:
same suggestion here, if it's a bug fix, should link some reference
For example, duckdb puts issue fix tests under a certain directory: https://github.com/duckdb/duckdb/tree/main/test/issues/general